lib/kitchen/driver/openstack.rb

```diff
@@ -303,14 +302,22 @@ def get_public_private_ips(server)
       end
 
       def get_ip(server)
-        unless config[:floating_ip].nil?
+        if config[:floating_ip]
           debug "Using floating ip: #{config[:floating_ip]}"
           return config[:floating_ip]
         end
+
+        # make sure we have the latest info
+        info 'Waiting for network information to be available...'
+        w = server.wait_for { reload && !addresses.empty? }
```

Comment:
:+1: I'd like to make the `:duration` a config option or override. In the case of @iceberg7